easier68k/core/opcodes/simhalt.py

```diff
@@ -0,0 +1,127 @@
+        """
+        # make the simulator halt
+        simulator.halt()
+
+        # increment the program counter
+        simulator.increment_program_counter(OpSize.LONG.value)
+
+    def __str__(self):
+        # Makes this a bit easier to read in doctest output
+        return 'SIMHALT command'
+
+    @classmethod
+    def command_matches(cls, command: str) -> bool:
+        """
+        Checks whether a command string is an instance of this command type
+        :param command: The command string to check (e.g. 'MOVE.B', 'LEA', etc.)
+        :return: Whether the string is an instance of this command type
+        """
+        return opcode_util.command_matches(command, 'SIMHALT')
+
+    @classmethod
+    def get_word_length(cls, command: str, parameters: str) -> int:
+        """
+        >>> Simhalt.get_word_length('SIMHALT', '')
+        2
+
+        Gets what the end length of this command will be in memory
+        :param command: The text of the command itself (e.g. "LEA", "MOVE.B", etc.)
+        :param parameters: The parameters after the command
+        :return: The length of the bytes in memory in words
+        """
+        valid, issues = Simhalt.is_valid(command, parameters)
+        if not valid:
+            return 0
+        # We can forego asserts in here because we've now confirmed this is valid assembly code
+
+        return 2
+
+    @classmethod
+    def is_valid(cls, command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        >>> Simhalt.is_valid('SIMHALT', '')[0]
+        True
+
+        >>> Simhalt.is_valid('SIMHALT.B', '')[0]
+        False
+
+        >>> Simhalt.is_valid('MOVE', '')[0]
+        False
+
+        >>> Simhalt.is_valid('SIMHALT', ' ')[0]
+        True
+
+        >>> Simhalt.is_valid('SIMHALT', 'D0')[0]
+        False
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        issues = []
+        try:
+            assert opcode_util.check_valid_command(command, 'SIMHALT', can_take_size=False), 'Command invalid'
+            assert not parameters.strip(), 'SIMHALT takes no parameters'
+
+            return True, issues
+        except AssertionError as e:
+            issues.append((e.args[0], 'ERROR'))
+            return False, issues
+
+    @classmethod
+    def disassemble_instruction(cls, data: bytearray) -> Opcode:
+        """
+        Parses some raw data into an instance of the opcode class
+
+        :param data: The data used to convert into an opcode instance
+        :return: The constructed instance or none if there was an error and
+            the amount of data in words that was used (e.g. extra for immediate
+            data) or 0 for not a match
+        """
+        return cls()
```

Comment:
Shouldn't this check if the instruction matches?

Reply:
cc @AdamuKaapan 

This was rebased, so that's why it shows up here. We probably should be checking that the instruction matches, or returning `None` if it doesn't.
Although I thought at one point we made the assumption that if the command matched (w/ another method), it would assume that disassembly should pass.

Reply:
Yeah, we've started going off the assumption of "the user/calling code already checked validity" for assembling instructions, I see no reason the same can't be applied to disassembly (assuming there is a method/check written for that, anyways).